buffer/src/main/java/io/netty/buffer/SlicedByteBuf.java

```diff
@@ -92,8 +86,13 @@ public boolean isDirect() {
     }
 
     @Override
-    public int capacity() {
-        return length;
+    public final int capacity() {
```

Comment:
@normanmaurer - note this change is technically breaking the interface by not allowing extension ... however I think allowing it to be overridden is a bug ... and making this final allows us to remove the `int length` ... which may be a big savings if there are a lot of sliced buffers